examples/apps/generic_client/generic_client.py

```diff
@@ -116,14 +115,203 @@ def ParseCommandLine(args):
+    requester_nonce = secrets.token_hex(16)
+    # Create work order params
+    wo_params = WorkOrderParams(
+        work_order_id, worker_id, workload_id, requester_id,
+        session_key, session_iv, requester_nonce,
+        result_uri=" ", notify_uri=" ",
+        worker_encryption_key=worker_encrypt_key,
+        data_encryption_algorithm="AES-GCM-256"
+    )
+    # Add worker input data
+    for value in in_data:
+        wo_params.add_in_data(value)
+
+    # Encrypt work order request hash
+    wo_params.add_encrypted_request_hash()
+
+    return wo_params
+
+
+def _create_work_order_receipt(wo_receipt, wo_params,
+                               client_private_key, jrpc_req_id):
+    # Create work order receipt object using WorkOrderReceiptRequest class
+    wo_request = json.loads(wo_params.to_jrpc_string(jrpc_req_id))
+    wo_receipt_request_obj = WorkOrderReceiptRequest()
+    wo_create_receipt = wo_receipt_request_obj.create_receipt(
+        wo_request,
+        ReceiptCreateStatus.PENDING.value,
+        client_private_key
+    )
+    logger.info("Work order create receipt request : {} \n \n ".format(
+        json.dumps(wo_create_receipt, indent=4)
+    ))
+    # Submit work order create receipt jrpc request
+    wo_receipt_resp = wo_receipt.work_order_receipt_create(
+        wo_create_receipt["workOrderId"],
+        wo_create_receipt["workerServiceId"],
+        wo_create_receipt["workerId"],
+        wo_create_receipt["requesterId"],
+        wo_create_receipt["receiptCreateStatus"],
+        wo_create_receipt["workOrderRequestHash"],
+        wo_create_receipt["requesterGeneratedNonce"],
+        wo_create_receipt["requesterSignature"],
+        wo_create_receipt["signatureRules"],
+        wo_create_receipt["receiptVerificationKey"],
+        jrpc_req_id
+    )
+    logger.info("Work order create receipt response : {} \n \n ".format(
+        wo_receipt_resp
+    ))
+
+
+def _retrieve_work_order_receipt(wo_receipt, wo_params, jrpc_req_id):
+    # Retrieve work order receipt
+    receipt_res = wo_receipt.work_order_receipt_retrieve(
+                        wo_params.get_work_order_id(),
+                        id=jrpc_req_id
+                    )
+    logger.info("\n Retrieve receipt response:\n {}".format(
+        json.dumps(receipt_res, indent=4)
+    ))
+    # Retrieve last update to receipt by passing 0xFFFFFFFF
+    jrpc_req_id += 1
+    receipt_update_retrieve = \
+        wo_receipt.work_order_receipt_update_retrieve(
+            wo_params.get_work_order_id(),
+            None,
+            1 << 32,
+            id=jrpc_req_id)
+    logger.info("\n Last update to receipt receipt is:\n {}".format(
+        json.dumps(receipt_update_retrieve, indent=4)
+    ))
+
+    return receipt_update_retrieve
+
+
+def _verify_receipt_signature(receipt_update_retrieve):
+    # Verify receipt signature
+    sig_obj = signature.ClientSignature()
+    status = sig_obj.verify_update_receipt_signature(
```

Comment:
Do need try-catch here?

Reply:
The  implementation of verify_update_receipt_signature( has try/catch. So here only function return value is checked.